pkg/hive/hive.go

```diff
@@ -104,14 +108,33 @@ func New(cells ...cell.Cell) *Hive {
 			ModulePrivateProviders: modulePrivateProviders,
 			ModuleDecorators:       moduleDecorators,
 			DecodeHooks:            decodeHooks,
-			StartTimeout:           5 * time.Minute,
-			StopTimeout:            1 * time.Minute,
-			LogThreshold:           100 * time.Millisecond,
+			StartTimeout:           defaults.HiveStartTimeout,
+			StopTimeout:            defaults.HiveStopTimeout,
+			LogThreshold:           defaults.HiveLogThreshold,
 		},
 		cells...,
 	)
 }
 
+func RegisterFlags(vp *viper.Viper, flags *pflag.FlagSet) {
+	flags.Duration(option.HiveStartTimeout, defaults.HiveStartTimeout, "Amount of time allocated for start hooks")
+	option.BindEnv(vp, option.HiveStartTimeout)
+
+	flags.Duration(option.HiveStopTimeout, defaults.HiveStopTimeout, "Amount of time allocated for stop hooks")
+	option.BindEnv(vp, option.HiveStopTimeout)
+
+	flags.Duration(option.HiveLogThreshold, defaults.HiveLogThreshold, "Optional threshold to reduce logging verbosity")
+	option.BindEnv(vp, option.HiveLogThreshold)
+}
```

Comment:
nit:

I would change descriptions of first two flags to something more descriptive like this:
- `Maximum time to wait for startup hooks to complete before timing out`
And for
"Optional threshold to reduce logging verbosity" I would remove optional word (most of parameters are optional anyway) and describe it like:
"Time limit after which a slow hook is logged"